camel/models/cohere_model.py

```diff
@@ -0,0 +1,240 @@
+        def safe_get(obj, *keys):
+            for key in keys:
+                if isinstance(obj, dict):
+                    obj = obj.get(key)
+                else:
+                    obj = getattr(obj, key, None)
+                if obj is None:
+                    return None
+            return obj
+
+        obj = ChatCompletion.construct(
+            id=unique_id,
+            choices=[
+                dict(
+                    index=0,
+                    message={
+                        "role": "assistant",
+                        "content": response.text,
+                    },
+                    finish_reason=safe_get(response, 'finish_reason'),
+                )
+            ],
+            created=safe_get(response, 'meta', 'created_at'),
+            model=self.model_type.value,
+            object="chat.completion",
+            usage={
+                "prompt_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'prompt_tokens'
+                )
+                or 0,
+                "completion_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'completion_tokens'
+                )
+                or 0,
+                "total_tokens": (
+                    (
+                        safe_get(
+                            response, 'meta', 'billed_tokens', 'prompt_tokens'
+                        )
+                        or 0
+                    )
+                    + (
+                        safe_get(
+                            response,
+                            'meta',
+                            'billed_tokens',
+                            'completion_tokens',
+                        )
+                        or 0
+                    )
+                ),
+            },
+        )
+        return obj
+
+    def _to_cohere_chatmessage(
+        self, messages: List[OpenAIMessage]
+    ) -> List[Dict[str, str]]:
+        new_messages = []
+        for msg in messages:
+            role = msg.get("role")
+            content = msg.get("content")
+
+            if role == "user":
+                new_messages.append({"role": "User", "message": content})
+            elif role == "assistant":
+                new_messages.append({"role": "Chatbot", "message": content})
+            elif role == "system":
+                new_messages.append({"role": "System", "message": content})
+            else:
+                raise ValueError(f"Unsupported message role: {role}")
+
+        return new_messages  # type: ignore[return-value]
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        """Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(
+                model=ModelType.GPT_3_5_TURBO
```

Comment:
use GPT_4O_MINI as default as GPT_3_5_TURBO will be depressed

```suggestion
                model=ModelType.GPT_4O_MINI
```